content/shared/influxdb3-processing-engine.md

```diff
@@ -0,0 +1,84 @@
+# Processing engine
+
+The Processing engine is an embedded Python virtual machine that runs inside an {{% product-name %}} database server. It executes Python code in response to triggers and database events without requiring external application servers or middleware.
+
+## How it works
+
+### Architecture
+
+The Processing engine runs Python code directly within a {{% product-name %}} server process. This design provides high performance and direct access to database resources.
+
+- **Embedded execution**: Code runs in the same process space as the database server
+- **Direct data access**: Zero-copy access to data
+- **Event-driven**: Responds to database writes, scheduled events, and HTTP requests
+- **Isolated contexts**: Maintains separation between different plugin executions
+- **Cache integration**: Access to system caches including Last values  and Distinct values 
+
+### Event processing flow
+
+When events occur in the database, the Processing engine handles them through a consistent sequence:
+
+1. A **trigger** activates the plugin based on one of three event types:
+   - Data writes to specific tables or all tables
+   - Scheduled events (time-based or cron expressions)
+   - HTTP requests to configured endpoints
+2. The engine loads the associated **plugin** specified in the trigger configuration
+3. The plugin receives context data specific to the trigger type:
+   - Write triggers: the written data and table information
+   - Schedule triggers: the scheduled call time
+   - HTTP triggers: the request object with methods, headers, and body
+4. The plugin processes the received data, can query the database, call external tools, and write results back
```

Comment:
```suggestion
4. When the plugin runs, it can process the received data and arguments, access the shared API through `influxdb3_local` for writing, querying, and caching data, and integrate with external systems.
```